common/oatbox/cache/SetupFileCache.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA ;
+ */
+
+declare(strict_types=1);
+
+namespace oat\oatbox\cache;
+
+use oat\generis\persistence\PersistenceManager;
+use oat\oatbox\service\ConfigurableService;
+
+class SetupFileCache extends ConfigurableService
+{
+    public const PERSISTENCE = 'cache';
+
+    public function createDirectory($cachePath): bool
+    {
+        if (is_dir($cachePath) && is_writable($cachePath)) {
+            return true;
+        }
+
+        return @mkdir($cachePath, 0700, true);
+    }
```

Comment:
I would throw an exception here instead and not return a boolean, since we always have to handle this boolean outside and throw an exception anyway. 

The exception can also contain better details about what happened where a boolean does not.

https://github.com/oat-sa/tao-core/pull/3561/files#r911034082

```suggestion
    public function createDirectory($cachePath): void
    {
        if (is_dir($cachePath) && is_writable($cachePath)) {
            return true;
        }

        if (!@mkdir($cachePath, 0700, true)) {
            throw new Exception(sprintf( 'Could not create application cache at "%s"', $cachePath) );
        }
    }
```